user-interface/src/main/java/life/qbic/datamanager/views/projects/create/BioIconComboboxFactory.java

```diff
@@ -0,0 +1,74 @@
+package life.qbic.datamanager.views.projects.create;
+
+import com.vaadin.flow.component.AbstractField.ComponentValueChangeEvent;
+import com.vaadin.flow.component.HasValue.ValueChangeListener;
+import com.vaadin.flow.component.combobox.ComboBox;
+import com.vaadin.flow.component.html.Span;
+import com.vaadin.flow.component.icon.AbstractIcon;
+import com.vaadin.flow.component.icon.VaadinIcon;
+import com.vaadin.flow.data.renderer.ComponentRenderer;
+import java.io.Serial;
+import java.util.List;
+import life.qbic.datamanager.views.projects.project.experiments.experiment.ExperimentDetailsComponent.BioIcon;
+
+/**
+ * Factory class for creating ComboBoxes allowing the selection of icons for species and specimen.
+ */
+public class BioIconComboboxFactory {
+
+  public ComboBox<BioIcon> iconBox(List<BioIcon> options, String title) {
+    ComboBox<BioIcon> comboBox = new ComboBox<>(title);
+    comboBox.setItems(options);
+    comboBox.setWidth("150px");
+    comboBox.setItemLabelGenerator(
+        BioIcon::getLabel);
+    comboBox.setRenderer(new ComponentRenderer<>(iconResource -> {
+      Span element = new Span();
+      element.addClassName("icon-and-component");
+      AbstractIcon<?> icon = iconResource.getIconResource().createIcon();
+      icon.addClassName("primary");
+      element.add(icon);
+      element.add(iconResource.getLabel());
+      return element;
+    }));
+    return comboBox;
```

Comment:
```suggestion
    comboBox.addValueChangeListener(valueChanged -> valueChanged.getSource()
        .setPrefixComponent(valueChanged.getValue().getIconResource().createIcon()));
    return comboBox;
```
You could also make the combobox change the prefix icon based on the selection 😉